sdk/storage/storage-file-datalake/test/pathclient.spec.ts

```diff
@@ -797,6 +798,248 @@ describe("DataLakePathClient", () => {
     );
   });
 
+  it("append with acquire lease", async () => {
+    const body = "HelloWorld";
+
+    const tempFileName = recorder.getUniqueName("tempfile2");
+    const tempFileClient = fileSystemClient.getFileClient(tempFileName);
+    const leaseId = "ca761232ed4211cebacd00aa0057b223";
+
+    await tempFileClient.create();
+
+    await tempFileClient.append(body, 0, body.length, {
+      proposedLeaseId: leaseId,
+      leaseDuration: 15,
+      leaseAction: "acquire",
+    });
+
+    let gotError = false;
+    try {
+      await tempFileClient.append(body, body.length, body.length, {
+        flush: true,
+      });
+    } catch (err) {
+      gotError = true;
+      assert.ok(
+        (err as any).message.startsWith(
+          "There is currently a lease on the resource and no lease ID was specified in the request."
+        )
+      );
+    }
+    assert.ok(gotError, "Should throw out an exception to write to a leased file without lease id");
+
+    await tempFileClient.append(body, body.length, body.length, {
+      conditions: {
+        leaseId: leaseId,
+      },
+      flush: true,
+    });
+
+    const properties = await tempFileClient.getProperties();
+    assert.equal(properties.contentLength, body.length * 2);
+    assert.equal(properties.leaseState, "leased");
+    assert.equal(properties.leaseDuration, "fixed");
+    assert.equal(properties.leaseStatus, "locked");
+
+    await sleep(15);
```

Comment:
To avoid test running time too long, could we only sleep in record mode, but not sleep on playback mode?
Besides that, as we already has leaseID, could we delete the file with leaseID, without wait 15s.

Reply:
Added lease condition for delete operation.

Reply:
Good for me!